jscomp/test/js_promise_basic_test.ml

```diff
@@ -0,0 +1,134 @@
+
+
+open Js_promise
+
+let assert_bool b =
+  if b then ()
+  else
+    raise (Invalid_argument "Assertion Failure.")
+
+let fail _ =
+  (* assert_bool false *)
+  assert false 
+
+let thenTest () =
+  let p = resolve 4 in
+  p |> then_ (fun x -> resolve @@ assert_bool (x = 4))
+
+let andThenTest () =
+  let p = resolve 6 in
+  p |> then_ (fun _ -> resolve (12))
+    |> then_ (fun y -> resolve @@ assert_bool (y = 12))
+
+let catchTest () =
+  let p = reject "error" in
+  p |> then_ fail
+    |> catch (fun error -> resolve @@ assert_bool (Obj.magic error == "error"))
```

Comment:
@bobzhang is there another version of `catch` that is more typed?
Btw not suggesting it should be the version for every user. This version is nice and simple.